src/stores/WalletHistoryStore.js

```diff
@@ -0,0 +1,63 @@
+import { observable, action } from 'mobx';
+import _ from 'lodash';
+
+import Transaction from './models/Transaction';
+import { queryAllTransactions } from '../network/graphQuery';
+import { SortBy, TransactionType } from '../constants';
+
+
+export default class WalletHistoryStore {
+  @observable fullList = []
+  @observable list = []
+  @observable orderBy = 'createdTime'
+  @observable direction = SortBy.Descending
+  @observable limit = 500
+  @observable skip = 0
+  @observable perPage = 10
+  @observable page = 0
+  @observable expanded = []
+
+  @action.bound
+  queryTransactions = async (orderBy = this.orderBy, direction = this.direction, limit = this.limit, skip = this.skip) => {
+    try {
+      const filters = [{ type: TransactionType.Transfer }];
+      const orderByObj = { field: orderBy, direction };
+
+      const result = await queryAllTransactions(filters, orderByObj, limit, skip);
+      this.fullList = _.map(result, (tx) => new Transaction(tx));
```

Comment:
we should have a `runInAction` wrapped around the 2 lines after line 26 since we're modifying an `@observable` inside an `async @action`

Reply:
basically we always want to do this after we `await` in an `async @action`

Reply:
I know... kinda weird, but this will keep from weird bugs happening (like the list and the fullList might not update)

Reply:
Also, typically I like to split my `api request` methods and my `setter methods`.
```js
@action.bound
async init () {
  ...
  const transactions = await this.fetchTransactions()
  runInaction(() => {
    this.fullList = transactions
    this.list = ...
  })
}

@action.bound
async loadMore() {
  const transactions = await this.fetchTransactions(this.skip, this.limit)
  runInaction(() => {
    this.fullList = transactions
    this.list = ...
  })
}

fetchTransactions(skip, limit) {
  return queryAllTransactions(...)
}
```
because sometimes we might need to make an `api call` in another method in our store

Reply:
added runInActions. for splitting the api request, I feel it makes sense to just have the `queryTransactions` call set the list since this page doesn't have pagination and it's only called once.